mypy/checkexpr.py

```diff
@@ -1061,15 +1061,54 @@ def check_op(self, method: str, base_type: Type, arg: Node,
             # method of the other operand instead.
             rmethod = self.get_reverse_op_method(method)
             arg_type = self.accept(arg)
+            base_arg_node = TempNode(base_type)
+            error_results = []  # type: List[Tuple[Tuple[Type, Type], MessageBuilder]]
```

Comment:
@gvanrossum I can make this simpler for now.  I was attempting to preserve the current behavior for now while in a follow up PR constructing a better error message when we fail detailing all of the methods that were considered and why they were not usable.  

I find the error messages from the current implementation to be somewhat confusing.  For example:

```py
class A(object):
  def __lt__(self, other: 'B'):
    pass
class B(object):
  pass
class C(object):
  def __gt__(self, other: 'B'):
    pass
A() < C() # E: Unsupported operand types for > ("C" and "A")
```

In this case I would expect to get an error message about `<`, not `>`


RE: 80 chars 😆